### PR TITLE
Stop overtrain hook

### PR DESCRIPTION
**Pull request by gnovis**

### CODE REVIEW

File: cxflow/hooks/stop_overtrain.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+Module with StopOvertrain hook which stops the training when a model is overtrained.
+"""
+import numpy as np
+from typing import List
+
+from . import AbstractHook, TrainingTerminated, AccumulateVariables
+from ..datasets import AbstractDataset
+
+
+class StopOvertrain(AccumulateVariables):
```
Comment (petrbel) [inline cxflow/hooks/stop_overtrain.py]: I suggest rethinking the hook name... What about `StopOnOverfit`? Or maybe mention the long/short windows?

File: cxflow/hooks/stop_overtrain.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+        :param long_term: count of last examples representing long training period
+        :param short_term: count of last examples representing short training period
+        :param stream: name of the processed stream
+        :param loss_var: name of the processed loss variable
+        :raise AssertionError: if ``long_term`` < ``short_term``
+        """
+        super().__init__(variables=[loss_var], **kwargs)
+
+        assert long_term >= short_term
```
Comment (petrbel) [inline cxflow/hooks/stop_overtrain.py]: missing assertion message

File: cxflow/hooks/stop_overtrain.py (hunk 1 above)
Comment (petrbel) [reply cxflow/hooks/stop_overtrain.py]: `StopOnPlateau`?

Comment (gnovis) [reply cxflow/hooks/stop_overtrain.py]: I like `StopOnPlateau`, thanks.

**Pull request merged**